lib/trino-parquet/src/main/java/io/trino/parquet/reader/ParquetBlockFactory.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.parquet.reader;
+
+import io.trino.spi.block.Block;
+import io.trino.spi.block.LazyBlock;
+import io.trino.spi.block.LazyBlockLoader;
+
+import java.io.IOException;
+import java.util.function.Function;
+
+import static com.google.common.base.Preconditions.checkState;
+import static java.util.Objects.requireNonNull;
+
+public class ParquetBlockFactory
+{
+    private final Function<Exception, RuntimeException> exceptionTransform;
```

Comment:
The reason to create ParquetBlockFactory inside ParquetReader is that it tracks `currentPageId` and uses that for validation during loading of blocks. If the block factory comes from outside as a parameter, we'd need to validate that it is in the right state rather than having it guaranteed on construction.
It also matches the way this is handled in orc reader now.